CRM/Civicase/APIHelpers/CustomFieldFilter.php

```diff
@@ -51,9 +51,11 @@ private static function joinCustomFieldQuery(CRM_Utils_SQL_Select $sql, array $c
     $value = $value['IN'] ?? $value;
     $table = $columnGroup[0];
     $column = $columnGroup[1];
+    $mysqlVersion = CRM_Core_DAO::executeQuery('SELECT VERSION() AS version;')->fetchValue();
+    $isMysql8 = version_compare($mysqlVersion, '8.0') !== -1;
 
-    $conditions = array_map(function ($param) use ($table, $column) {
-      return "custom_case_to_{$table}.{$column} REGEXP '[[:<:]]" . $param . "[[:>:]]'";
+    $conditions = array_map(function ($param) use ($table, $column, $isMysql8) {
+      return $isMysql8 ? "custom_case_to_{$table}.{$column} REGEXP '\\b" . $param . "\\b'" : "custom_case_to_{$table}.{$column} REGEXP '[[:<:]]" . $param . "[[:>:]]'";
```

Comment:
@shahrukh-compuco is there a generic solution that doesn't require checking the MySQL version?

Reply:
unfortunately no as that is a backward incompatible change
so for now we will have to live with this version check solution until we move to mysql 8 completely